src/main/java/com/meronat/iplog/commands/HistoryCommand.java

```diff
@@ -57,7 +57,7 @@ public CommandResult execute(CommandSource src, CommandContext args) throws Comm
 
         Optional<InetAddress> optionalAddress = args.getOne("ip");
 
-        if (optionalUser.isPresent() && optionalAddress.isPresent()) {
+        if (!optionalUser.isPresent() && !optionalAddress.isPresent()) {
```

Comment:
This exception is if they specify both so I believe there should be not `!` for either of these, as this case is handled in the `else` on line 127 below.